### ISSUE TITLE
[security] Confirmation before performing Autotype

### ISSUE THREAD

**Issue by notoriousturtle** [MacPass/MacPass#934]
**Is your feature request related to a problem? Please describe.**
Display the about to be typed credentials info window (like you see when there are multiple matches) before performing the auto type. 

This will prevent you from accidentally sending credentials to a website where there has been a bad match.

**Comment by mstarke** [comment]
This is a feature I was pondering about as well. My initial idea was to use the confirmation only in password inputs and then display an overlay directly where the text would be typed (e.g. by getting the focus widget via the accessovilty API and then overlaying a confirmation box) but I haven’t had the time to implement it. But this is certainly something worth adding as is adds a viable layer of security.

**Comment by georgesnow** [comment]
I believe I saw was on the  keepassxc forum about them implementing using an auto type option of {COPY: field}. This would copy that field to the clipboard. Not standard function for the overall keepass database but it would work nicely. I would say I would love a feature like that I have been using the delay but sometimes it’s to long or short.

**Comment by georgesnow** [comment]
Oops my suggestion is a little divergent from the core request.

**Comment by georgesnow** [comment]
https://github.com/keepassxreboot/keepassxc/issues/2215

**Comment by notoriousturtle** [comment]
@georgesnow yep, that ones not exactly what I was looking for. Whilst it could work, I'd prefer to know the match was 100% before accessing the password. 
It also opens up another security issue. If you copy to clipboard, and your target site is reading your clipboard, well... :/

**Comment by georgesnow** [comment]
what about along those lines of autotype having {PROMPT: Field}? Again of course not the standard for KeePass database. I mention it cause I had tried to mimic it with pick fields, but AutoType doesn't prompt "in the middle" of the action. it only asks you at the beginning. 

The idea being AutoType would pause and prompt and wait for the user to select OK. Showing the field that its providing and from what entry.

**Comment by mstarke** [comment]
Both ideas are nice and could be considered for more plugins since this way the API is getting more mature and allows for better expansions. I would discourage extend this via core MacPass to better separate the supported feature set.

Proposal for two Plugins: Autotype Copy ‘’’{COPY}’’’ and Autotype prompt ‘’’{PROMPT:message}’’’

**Comment by georgesnow** [comment]
yeah i agree a plugin for the functionality seems more appropriate.
